src/main/java/io/supertokens/storage/mysql/queries/PasswordlessQueries.java

```diff
@@ -0,0 +1,539 @@
+            pst.setString(1, codeId);
+            pst.executeUpdate();
+        }
+    }
+
+    public static void createUser(Start start, UserInfo user)
+            throws StorageTransactionLogicException, StorageQueryException {
+        start.startTransaction(con -> {
+            Connection sqlCon = (Connection) con.getConnection();
+            try {
+                {
+                    String QUERY = "INSERT INTO " + Config.getConfig(start).getUsersTable()
+                            + "(user_id, recipe_id, time_joined)" + " VALUES(?, ?, ?)";
+                    try (PreparedStatement pst = sqlCon.prepareStatement(QUERY)) {
+                        pst.setString(1, user.id);
+                        pst.setString(2, RECIPE_ID.PASSWORDLESS.toString());
+                        pst.setLong(3, user.timeJoined);
+                        pst.executeUpdate();
+                    }
+                }
+
+                {
+                    String QUERY = "INSERT INTO " + Config.getConfig(start).getPasswordlessUsersTable()
+                            + "(user_id, email, phone_number, time_joined)" + " VALUES(?, ?, ?, ?)";
+                    try (PreparedStatement pst = sqlCon.prepareStatement(QUERY)) {
+                        pst.setString(1, user.id);
+                        pst.setString(2, user.email);
+                        pst.setString(3, user.phoneNumber);
+                        pst.setLong(4, user.timeJoined);
+                        pst.executeUpdate();
+                    }
+                }
+                sqlCon.commit();
+            } catch (SQLException throwables) {
+                throw new StorageTransactionLogicException(throwables);
+            }
+            return null;
+        });
+    }
+
+    public static void updateUserEmail_Transaction(Start start, Connection con, String userId, String email)
+            throws SQLException, UnknownUserIdException {
+        String QUERY = "UPDATE " + Config.getConfig(start).getPasswordlessUsersTable()
+                + " SET email = ? WHERE user_id = ?";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, email);
+            pst.setString(2, userId);
+            if (pst.executeUpdate() != 1) {
+                throw new UnknownUserIdException();
+            }
+        }
+    }
+
+    public static void updateUserPhoneNumber_Transaction(Start start, Connection con, String userId, String phoneNumber)
+            throws SQLException, UnknownUserIdException {
+        String QUERY = "UPDATE " + Config.getConfig(start).getPasswordlessUsersTable()
+                + " SET phone_number = ? WHERE user_id = ?";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, phoneNumber);
+            pst.setString(2, userId);
+            if (pst.executeUpdate() != 1) {
+                throw new UnknownUserIdException();
+            }
```

Comment:
You should make this function return the number of rows updated. And in start.java, we check if it's not 1, then we throw `UnknownUserIdException`. This is cause this file is only responsible for running the queries and not mapping the query result to the exceptions thrown from the interface functions.